.gitignore

```diff
@@ -51,3 +51,9 @@ docs/html
 
 # VSCode java extensions
 .project
+
+#clangd index files
```

Comment:
> Since `controller/java` doesn't depend on `platform/android`, I think the CHIPTool handles too much logic trying to bridge between the two libraries.
> 
> For example, in BluetoothManager (app code), we manually call the platform:
> 
> ```
> connectionId = AndroidChipPlatform.getInstance().bleManager.addConnection(this)
> ```
> 
> This forces us, in turn, to expose a `connectionId` field in `ChipDeviceController.pairDevice()`. If we can instead make `controller/java` depend on `platform/android`, then we can inline this logic into the `pairDevice` function.
> 
> Note that the Java GN rules don't currently support this, but after #9876 merges, it will work.

I think this because of the work flow, let's compare it with chip-tool linux:
1. [both] use discover command to search Ble Advertising with target discriminator
2. [android] connect to the ble device
3. [android] PairDevice with a connected connectionId
3. [Linux] PairDevice with discriminator
4. [Linux] connect to the ble device
5. other

If we use PairDevice with discriminator instead of PairDevice with a connected connectionId, I think there will less logic in CHIPTool. Do you know why we use different flow in Android and Linux platform?

Reply:
After that, CHIPTool will only init platform/Android with platform private apis, and ChipDeviceController will only use platform/Android with platform public apis.

Reply:
> nit: add a space after the `#` (should do this everywhere in this PR)

fixed at 21c564d

Reply:
#9876 has merged. Yes, my end goal is to be able to have PairDevice(discriminator), and put the connectionId logic inside ChipDeviceController. Basically, for the controller library to be able to talk to the platform library.

I think the flows were previously different because Android didn't have platform layer, but now that we do, they can follow the same flow.

Reply:
agree, while hope we can do it after this PR, really lots of changes in this PR and lots of conflicts every rebase

Reply:
Ok, I'm fine with doing this in a follow-up.